public/components/markdown_with_blink_cursor.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiMarkdownFormat, EuiMarkdownFormatProps } from '@elastic/eui';
+import React from 'react';
+import { useRef } from 'react';
+import { useEffect } from 'react';
+
+function getLastTextNode(node: HTMLElement | Node): Node | null | undefined {
+  if (node.nodeType === node.TEXT_NODE) {
+    return node;
+  }
+  const children = node.childNodes;
+  for (let i = children.length - 1; i >= 0; i--) {
+    const child = children[i];
+    const result = getLastTextNode(child);
+    if (result) {
+      return result;
+    }
+    return null;
+  }
+}
+
+export const MarkdownWithBlinkCursor = (props: EuiMarkdownFormatProps & { loading?: boolean }) => {
+  const ref = useRef<HTMLDivElement | null>(null);
```

Comment:
Minor: I think `| null` here is unnecessary since HTMLDivElement already includes type `null`.

Reply:
Sure, updated.